src/scripts/main.js

```diff
@@ -1,3 +1,172 @@
+    if (e.target.tagName === 'TD') {
+      const row = e.target.parentNode;
+
+      if (selectedRow) {
+        selectedRow.classList.remove('active');
+      }
+      selectedRow = row;
+      row.classList.add('active');
+    }
+  });
+
+  const formHtml = `
+    <form class="new-employee-form">
+      <label>Name: <input name="name" type="text" data-qa="name"></label>
+      <label>Position: <input name="position" type="text" data-qa="position"></label>
+      <label>Office:
+        <select name="office" data-qa="office">
+          <option>Tokyo</option>
+          <option>Singapore</option>
+          <option>London</option>
+          <option>New York</option>
+          <option>Edinburgh</option>
+          <option>San Francisco</option>
+        </select>
+      </label>
+      <label>Age: <input name="age" type="number" data-qa="age"></label>
+      <label>Salary: <input name="salary" type="number" data-qa="salary"></label>
+      <button type="button" id="save-button">Save to table</button>
+    </form>
+  `;
+
+  document.body.insertAdjacentHTML('beforeend', formHtml);
+
+  const form = document.querySelector('.new-employee-form');
+  const saveButton = document.getElementById('save-button');
+
+  saveButton.addEventListener('click', () => {
+    const employeeName = form.name.value.trim();
+    const position = form.position.value.trim();
+    const office = form.office.value;
+    const age = form.age.value;
+    const salary = form.salary.value;
+
+    if (employeeName.length < 4) {
+      showNotification(
+        'Error',
+        'Name must be at least 4 characters long',
+        'error',
+      );
+
+      return;
+    }
+
+    if (age < 18 || age > 90) {
+      showNotification('Error', 'Age must be between 18 and 90', 'error');
+
+      return;
+    }
+
+    if (!employeeName || !position || !office || !age || !salary) {
+      showNotification('Error', 'All fields are required', 'error');
+
+      return;
+    }
+
+    const formattedSalary = `$${Number(salary).toLocaleString('en-US')}`;
+
+    const newRow = document.createElement('tr');
+
+    newRow.innerHTML = `
+      <td>${employeeName}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${formattedSalary}</td>
+    `;
+    employeeTable.querySelector('tbody').appendChild(newRow);
+
+    showNotification('Success', 'Employee added to the table', 'success');
+    form.reset();
+  });
+
+  function showNotification(title, message, type) {
+    const notification = document.createElement('div');
+
+    notification.className = `notification ${type}`;
+    notification.setAttribute('data-qa', 'notification');
+
+    notification.innerHTML = `
+      <span class="title">${title}</span>
+      <p>${message}</p>
+    `;
+    document.body.appendChild(notification);
+
+    setTimeout(() => {
+      notification.remove();
+    }, 3000);
+  }
+
+  employeeTable.addEventListener('dblclick', function (e) {
```

Comment:
The `dblclick` event is not very user-friendly on mobile devices. Consider using a more mobile-friendly event.